### PR TITLE
Request All Indexed Attestations Since Genesis in Slasher on Startup

### PR DESCRIPTION
**Pull request by rauljordan**
Part of #4836 

---

# Description

**Write why you are making the changes in this pull request**

Slasher needs to get all indexed attestation data from a beacon node upon startup, namely the logic implemented in #4892.

**Write a summary of the changes you are making**

This PR adds a function that can request all indexed attestations in a beacon node by epoch in the slasher detection service. On startup, we get the chain head of the beacon node and begin requesting all these indexed attestations from genesis up to the current epoch of the beacon node. Concurrently, the slasher process is also listening to streamed attestations and blocks from the beacon node to ensure no data is missed.

**Link anything that would be helpful or relevant to the reviewers**

This PR forces slasher to wait until the beacon node is fully synced before starting slashing detection by polling the `GetSyncStatus` endpoint of the beacon node every `syncStatusPollingInterval` (set as a default of seconds per slot).

### CODE REVIEW

File: slasher/beaconclient/chain_data_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/mock/gomock"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	logTest "github.com/sirupsen/logrus/hooks/test"
+
```
Comment (rauljordan) [inline slasher/beaconclient/chain_data_test.go]:
```suggestion
```

File: slasher/beaconclient/historical_data_retrieval.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package beaconclient
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"go.opencensus.io/trace"
+
```
Comment (rauljordan) [inline slasher/beaconclient/historical_data_retrieval.go]:
```suggestion
```

File: slasher/detection/service.go [hunk 3]
```diff
@@ -6,7 +6,9 @@ import (
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
```
Comment (rauljordan) [inline slasher/detection/service.go]: Review this file

File: slasher/beaconclient/chain_data.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"context"
+	"time"
+
+	ptypes "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"go.opencensus.io/trace"
+
```
Comment (terencechain) [inline slasher/beaconclient/chain_data.go]:
```suggestion
```

**Pull request merged**